### PR TITLE
Change relations to many to many

### PR DESCRIPTION
**Pull request by JustinD85**
- Initially, we had our setup working for a one to many relationship. Upon reading the many-to-many lesson I noticed we no longer need the one-to-many relation if we setup a many-to-many. In fact, the one-to-many was causing major headaches and so they were removed.
- Lesson followed: http://backend.turing.io/module2/lessons/many_to_many
- Verified working steps:
  - bob = Author.create(name:"Bob")
  - bob.books.create(*args here)
  - Book.last.authors
  - Author.last.books
- Also, I added some validations but am unsure if they are necessary. I'll mark them in the code below.

### CODE REVIEW

File: app/models/author_book.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class AuthorBook < ApplicationRecord
-belongs_to :author
-belongs_to :book
-end
\ No newline at end of file
+ belongs_to :author
+ belongs_to :book
+
```
Comment (JustinD85) [inline app/models/author_book.rb:4]: Should we have this validation?

File: spec/models/author_book_spec.rb [hunk 2]
```diff
@@ -1,14 +1,17 @@
 require 'rails_helper'
 
 RSpec.describe AuthorBook do
-  # describe "validations" do
-  #   it { should validate_presence_of :book }
-  #   it { should validate_presence_of :author }
-  # end
+  #Not sure if validations are necessary
```
Comment (JustinD85) [inline spec/models/author_book_spec.rb:4]: Should we have this check for validation?

File: app/models/author_book.rb (hunk 1 above)
Comment (Martsy) [reply app/models/author_book.rb:4]: Yes, after going through the  lesson this morning. I can confirm we do need the code.  The lesson also does have the following: 

class PlaylistSong < ApplicationRecord
  belongs_to :playlist
  belongs_to :song
end

File: spec/models/author_book_spec.rb (hunk 2 above)
Comment (Martsy) [reply spec/models/author_book_spec.rb:4]: Based on keeping the above code/relationship we should keep this test for coverage and verification

**Comment by Martsy** [comment]
Great job finding the error.  The lesson we had this morning cleared up many of the foreign/primary key topics I was confused about.

**Pull request merged**